File: C++/476. Number Complement.cpp
class Solution {
public:
    int bitwiseComplement(int n) {
        if(n==0)
        return 1;
        int i=0,r=0;
        string s="";
        while(n!=0){
            if(n%2==0)
            s+='1';
            else
            s+='0';
            n/=2;
        }
        for(char c:s){
            if(c=='1')
            r+=pow(2,i);
            i++;
        }
        return r;
    }
};

File: JAVA/476. Number Complement.java
class Solution {
    public int findComplement(int n) {
        int r=0;
        String s="";
        while(n!=0){
            if(n%2==0)
            s+='1';
            else
            s+='0';
            n/=2;
        }
        for(int j=0;j<s.length();j++){
            if(s.charAt(j)=='1')
            r+=Math.pow(2,j);
        }
        return r;
    }
}